cmd/clusterctl/client/client.go

```diff
@@ -40,6 +40,9 @@ type Client interface {
 	// GetClusterTemplate returns a workload cluster template.
 	GetClusterTemplate(options GetClusterTemplateOptions) (Template, error)
 
+	// GetKubeconfig returns the kubeconfig
+	GetKubeconfig(options GetKubeconfigOptions) error
```

Comment:
```suggestion
	GetKubeconfig(options GetKubeconfigOptions) (string, error)
```
The responsibility to print the output should not be in the library